pkg/yamlprocessor/yamlprocessor.go

```diff
@@ -35,63 +51,62 @@ import (
 // [num] : object/element of array by number
 //
 // e.g. "$.foo.bar[0].baz"
-func GetValue(yml []byte, path string) (interface{}, error) {
-	if len(yml) == 0 {
-		return nil, fmt.Errorf("empty yaml given")
-	}
+func (p *Processor) GetValue(path string) (interface{}, error) {
 	if path == "" {
 		return nil, fmt.Errorf("no path given")
 	}
 
-	p, err := goyaml.PathString(path)
+	yamlPath, err := goyaml.PathString(path)
+	if err != nil {
+		return nil, fmt.Errorf("failed to parse path %s: %v", path, err)
+	}
+
+	node, err := yamlPath.FilterFile(p.file)
 	if err != nil {
 		return nil, err
 	}
 
 	var value interface{}
-	if err := p.Read(bytes.NewReader(yml), &value); err != nil {
+	if err := goyaml.Unmarshal([]byte(node.String()), &value); err != nil {
 		return nil, err
 	}
+
 	return value, nil
 }
 
-// ReplaceValue replaces the value placed at a given path with
-// a given value, and then gives back the new yaml bytes.
-//
-// For available operators for the path, see GetValue().
-func ReplaceValue(yml []byte, path string, value string) ([]byte, error) {
-	if len(yml) == 0 {
-		return nil, fmt.Errorf("empty yaml given")
-	}
+// ReplaceString replaces the value placed at a given path with
+// a given string value.
+func (p *Processor) ReplaceString(path, value string) error {
 	if path == "" {
-		return nil, fmt.Errorf("no path given")
+		return errors.New("no path given")
 	}
 
-	p, err := goyaml.PathString(path)
+	yamlPath, err := goyaml.PathString(path)
 	if err != nil {
-		return nil, err
-	}
-	file, err := parser.ParseBytes(yml, parser.ParseComments)
-	if err != nil {
-		return nil, err
+		return fmt.Errorf("failed to parse path %s: %w", path, err)
 	}
 
 	// Retrieve the current node placed at the specified path.
-	oldNode, err := p.FilterFile(file)
+	oldNode, err := yamlPath.FilterFile(p.file)
 	if err != nil {
-		return nil, err
+		return err
 	}
+
 	newNode := &ast.StringNode{
 		BaseNode: &ast.BaseNode{},
 		Token:    oldNode.GetToken(),
 		Value:    value,
 	}
 
-	err = p.ReplaceWithNode(file, newNode)
+	return yamlPath.ReplaceWithNode(p.file, newNode)
+}
+
+func (p *Processor) Bytes() ([]byte, error) {
```

Comment:
nit: It's more effective to give back `io.Reader` because it can prevent us from unneeded allocation. In case that we need string, I'm gonna say just calling `String()` enough.

I understand just calling `Bytes` is a bit simpler, so it's up to you.

```go
func (p *Processor) Reader() io.Reader {
	return p.file
}

func (p *Processor) String() string {
	return p.file.String
}
```

Reply:
Nice catch! Copied from the old code.
Using `String` directly looks good. 👍

Reply:
Fixed this!